runtime/stdlib.js

```diff
@@ -352,85 +352,129 @@ function caml_compare_val (a, b, total) {
   var stack = [];
   for(;;) {
     if (!(total && a === b)) {
-      if (a instanceof MlBytes) {
-        if (b instanceof MlBytes) {
+      // Compute TAG for [a]
+      var ta = 1001;
+      if (typeof a === "number") ta = 1000;
```

Comment:
The magic numbers in this code make it really hard to follow.  Could these be pulled out into variables?

Reply:
my fear was that using variable in `switch` cases could be slower.
Theses tags are the ocaml tags, with small additions
```
let lazy_tag = 246
let closure_tag = 247
let object_tag = 248
let infix_tag = 249
let forward_tag = 250

let no_scan_tag = 251

let abstract_tag = 251
let string_tag = 252
let double_tag = 253
let double_array_tag = 254
let custom_tag = 255
let final_tag = custom_tag


let int_tag = 1000
let out_of_heap_tag = 1001
let unaligned_tag = 1002
```

Additions are
```
1255 (similar to custom_tag 255)
1256 (for JavaScript objects that have a compare method)  
1247 (similar to closure_tag 1247)
```